common/core/gpudevice.cpp

```diff
@@ -201,6 +201,8 @@ void GpuDevice::DisplayManager::HotPlugEventHandler() {
           "UpdateDisplayState.");
       UpdateDisplayState();
     }
+    /* Adding sleep to avoid busy looping before a retry */
+    usleep(1000);
```

Comment:
Ok, in that case a loop is required so lets keep it with a sleep. We won't need sleep even if there was one blocking system call inside the loop but here that's not the case, that read() is non-blocking and will wait till it reads bytes or fails.